syntax = "proto3";
package helloworld;

// The greeting service definition.
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}

//message getRequest{
//  int32 versionNumber = 1;
//  int64 requestSequenceNumber = 2;
//  string key = 3;
//}
//
//message getResponse{
//  int32 versionNumber = 1;
//  int64 requestSequenceNumber = 2;
//  string key = 3;
//  string value = 4;
//  int64 commitedSequenceNumber = 5;
//}
//
//message putRequest{
//  int32 versionNumber = 1;
//  int64 requestSequenceNumber = 2;
//  string key = 3;
//}
//
//message putReponse{
//  int32 versionNumber = 1;
//  int64 requestSequenceNumber = 2;
//  int64 putSequenceNumber = 3;
//}
//
//message commitRequest{
//  int32 versionNumber = 1;
//  int64 requestSequenceNumber = 2;
//  int64 commitIndex = 3;
//  string key = 4;
//  string value = 5;
//}
//
//message commitResponse{
//  int32 versionNumber = 1;
//  int64 requestSequenceNumber = 2;
//  int32 status = 3;
//}